ci(code-cov): adding code-cov to project

sethkfman opened the pull request:
<!--
Thanks for your contribution!

Please ensure that any applicable requirements below are satisfied before submitting this pull request. This will help ensure a quick and efficient review cycle.
-->

**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Adding GH workflow code to add code-cov. This is the initial configuration and will require a bit more fine tuning once main has been updated with the workflow.

This is to support a more flexible coverage tool and complement SonarCloud static analysis. 

**Screenshots/Recordings**

NA

**Checklist**

* [NA] There is a related GitHub issue
* [NA] Tests are included if applicable
* [x] Any added code is fully documented

sethkfman committed:
added code cov action

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
update ignore patterns for test coverage

sethkfman committed:
add codecov.yml file

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
update invalid codecov.yml file

sethkfman committed:
added range and percision codecov.yml file

sethkfman committed:
updated excluded files

sethkfman committed:
add ci pass required

sethkfman committed:
bump threshold

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
added carriage return to codecov config file

Gudahtt commented on codecov.yml:
```diff
@@ -0,0 +1,21 @@
+      default: # This can be anything, but it needs to exist as the name
+        # basic settings
+        target: auto
+        threshold: 32%
```
Have you considered setting thresholds in Jest instead, and using that as the quality gate? That's what the extension (and basically all of our other repos) do. The extension uses codecov just for better coverage reports (and used to use it for better status checks as well, though that has been temporarily disabled due to an extension-specific bug in how that integration is setup)

sethkfman replied:
I have enabled the global code coverage in Jest. I see extension has a lot more code to parallel test runs and evaluate coverage. I will exclude that work on this PR. 

The current workflow would require devs to look at a failing result of test cases update. You can see an example here: https://github.com/MetaMask/metamask-mobile/actions?query=branch%3Aci%2Fadd-code-cov

sethkfman committed:
updated global coverage

sethkfman committed:
remove codecov

sethkfman committed:
setup config and baseline

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
update threshold

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

Gudahtt commented on .github/workflows/ci.yml:
```diff
@@ -69,6 +69,10 @@ jobs:
       - run: yarn test:unit --forceExit --silent
         env:
           NODE_OPTIONS: --max_old_space_size=20480
+      - name: Upload coverage reports to Codecov
```
It looks like this step wouldn't run in the case where coverage was too low. The report would be helpful then in showing where to increase coverage. You could ensure that this always runs by doing something like this:

```suggestion
      - name: Upload coverage reports to Codecov
        if: ${{ always() }}
```

Not sure how that would act if it failed during setup though, or if no coverage results exist. It might handle it gracefully.

Gudahtt reviewed (dismissed):
LGTM! One pending suggestion but this still seems like a step forward regardless

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

Gudahtt committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
remove token due to public repo

sethkfman committed:
Update .github/workflows/ci.yml

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

Cal-L committed:
fix: #893 fix immediate lock timer (#6653)

* Stage saga changes

* Create action creators for authentication state machine

* Remove unused code

* Start and end state machine

* Clean up LockScreen

* Clean up LockManager

* Remove unused code

* Notify auth state machine from Authentication service

* Organize auth state machine in sagas

* Change sagas to TS

* Remove unused changes

* Add type to navigation in NavigationService

* Clean up sagas

* Remove console log

* Add tests for sagas

* Remove password set

* Fix saga lint issues

* Fix missing deps

* Remove ts ignore from sagas

* Clarify navigation action

* Remove commented logger

* Create LOCK_SCREEN route const

* Provide comments to sagas

* Rename biometrics listener

* set timer immediately v2

* Add biometricsStateMachineId to auth code

* Remove unused user actions

* Clean up LockManager logic to work with Android and iOS

* Better handle authentication sagas

* Fix saga export error

* Fix saga unit tests

* Bump version for test

* Refine logout state when using biometrics

* Bump version

* Remove retry logic for lock screen

* Cancel biometrics state machine when logging out

* Clean up lock screen

* Reset isAuthenticating when biometrics fail in SecureKeychain.js

* Move LockScreen to top nav level

* Persist app state when LockManager shows.

* Fix saga tests

* Move LockManager initialization

* Use existing login and logout actions for biometric state machine

* Bump versions

* Use replace when logging out

* Update LockScreen snapshoit

* Comment bioStateMachineId

* Remove LockManager to LockManagerService

* Add comments to Authentication service

* Convert LockManagerService to Typescript

* Fix typo

* Update LockManagerService visibility modifiers

* Fix lint issues

---------

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: tommasini <tommasini15@gmail.com>

github-actions[bot] committed:
7.4.0 (#6806)

* 7.4.0

* updated format changelog

* fix: define ReactNativePayments podspec path in podfile (#6821)

* from address changing on transaction object (#6841)

* fix: invalid transaction data used for approve transaction (#6833) (#6842)

* use valid txn data when approving txn

* rename variable

Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>

* add check approval request exists before rejecting (#6838)

* build 1155 RC2 (#6861)

* fix(test): Permission Systems fix taping connect modals (#6868)

* fix tapping connect modals

* fixed lint issues

* removed unused function, not referenced anywhere

* chore: merge 7.3.1 cherry-pick (#6923)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

* chore: version bump for 7.4.0 RC 3 (#6924)

* build 1159 RC4 (#6928)

---------

Co-authored-by: metamaskbot <metamaskbot@users.noreply.github.com>
Co-authored-by: sethkfman <Seth.Kaufman@consensys.net>
Co-authored-by: Pedro Pablo Aste Kompen <wachunei@gmail.com>
Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: Sylva Elendu <iamsylvaelendu@gmail.com>
Co-authored-by: Vinicius Stevam <45455812+vinistevam@users.noreply.github.com>
Co-authored-by: Chris Wilcox <chris.wilcox@consensys.net>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump word-wrap from 1.2.3 to 1.2.4 (#6840)

Bumps [word-wrap](https://github.com/jonschlinkert/word-wrap) from 1.2.3 to 1.2.4.
- [Release notes](https://github.com/jonschlinkert/word-wrap/releases)
- [Commits](https://github.com/jonschlinkert/word-wrap/compare/1.2.3...1.2.4)

---
updated-dependencies:
- dependency-name: word-wrap
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>
Co-authored-by: legobeat <109787230+legobeat@users.noreply.github.com>

Gudahtt committed:
chore: remove unused Jest preprocessor (#6913)

The file `jest.preprocessor.js` has been unused since #6170.

gantunesr committed:
docs: Update QA section in PR template (#6930)

SamuelSalas committed:
fix: Remove splash animation wait step from Cold Start Launch time test script (#6931)

* Change wait for splash animation to expect to displayed assertion

* Remove splash animation steps

tommasini committed:
feat: Translation (#6937)

vinistevam committed:
remove onBackdropPress for signatures (#6936)

sleepytanya committed:
feat: Migrate wdio 'SendEthMultisig' test to Detox (#6934)

gauthierpetetin committed:
feat: github action to check if PR has requested labels before being merged (#6794)

* feat(action): check if pr includes requested labels

* fix(template): adjust PR temaple

* fix(action): add missing QA label

* fix(action): check if no label prevents merging

* fix(action): remove QA label check

* fix(action): add missing reopened condition

* fix(action): increase list of labels which prevent merges

* fix(action): add labeling guidelines

* fix(action): typo

* fix(action): typo

sleepytanya committed:
fix: update README.md (#6853)

sleepytanya committed:
Restore deleted SendEthMultisig.feature file (#6956)

brianacnguyen committed:
fix: custom position logic for badgeWrapper (#6864)

* Fixed custom position logic for badgeWrapper

* Fixed lint error

brianacnguyen committed:
Added README to Avatar (#6895)

* Added README to Avatar

* Updated readme based on comments

* Fixed reference link for AvatarAccount

brianacnguyen committed:
style: Updated minor checkbox style (#6942)

* Updated minor checkbox changes

* Updated MultiSelect snapshot

NicolasMassart commented on jest.config.js:
```diff
@@ -15,7 +15,22 @@ const config = {
   // This is an environment variable that can be used to execute logic only in development
   collectCoverage: process.env.NODE_ENV !== 'production',
-  coveragePathIgnorePatterns: ['/node_modules/', '__mocks__', '<rootDir>/e2e/'],
+  coveragePathIgnorePatterns: [
```
Q: Do we also have to exclude `.github` folder? Not sure if scripts in CI are included in coverage.

Gudahtt replied:
Ah, thanks for noticing this! We don't need to include `.github` because Jest by default only collects coverage from modules that are run by Jest. But that itself is a problem because we're missing files not touched yet by Jest at all.

We should be setting `collectCoverageFrom`: https://jestjs.io/docs/configuration#collectcoveragefrom-array
Then we only have to worry about excluding modules in the range of files we're collecting coverage from (and those executed by Jest in the course of running tests, such as `node_modules`)

sethkfman replied:
Good improvement! I have updated and confirmed values are still accurate. @Gudahtt

vinistevam committed:
feat: add support for template and header on result pages (#6832)

sleepytanya committed:
Fix: add missed parenheses in the function call (#6958)

tommasini committed:
translation ens ipfs alert (#6966)

mmenoxx committed:
chore: small refactor (#6963)

* small refacto

* fix(): fixed linter errors

* fix(): fixed linter errors

---------

Co-authored-by: sethkfman <10342624+sethkfman@users.noreply.github.com>

segun committed:
feat: blockaid what's new popup (#6901)

* Add blockaid what's new

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* New image for both dark/light mode

* image should be first
fix podfile

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* What's new fix navigation

* Revert Podfile

* use Routes

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* revert podfile.lock

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Show what's new only if blockaid is enabled in settings.

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

segun committed:
feat: Add security alerts settings to experimental tab (#6779)

* Add security alerts settings to experimental tab

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* add more providers label

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* change styles label

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Fix tests

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Sort imports

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* use environment var to determine if to show blockaid or not

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

* Removed Podfile changes

---------

Signed-off-by: Olusegun Akintayo <akintayo.segun@gmail.com>

sethkfman committed:
updated jest config to have an allow list instead of eclude list for test coverage

sethkfman committed:
bumped thresholds

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

Gudahtt commented on jest.config.js:
```diff
@@ -15,7 +15,16 @@ const config = {
   snapshotSerializers: ['enzyme-to-json/serializer'],
   // This is an environment variable that can be used to execute logic only in development
   collectCoverage: process.env.NODE_ENV !== 'production',
-  coveragePathIgnorePatterns: ['/node_modules/', '__mocks__', '<rootDir>/e2e/'],
```
I think we'll still want to ignore `__mocks__`. There are some examples of that present in the `app` directory`.

We might want to ignore `__snapshots__` too, I see that around

Gudahtt replied:
Oh I see, the snapshots would be excluded due to their file extensions, disregard that idea.

The mocks are JavaScript/TypeScript though, so that still seems valid to include.

sethkfman committed:
Merge branch 'main' into ci/add-code-cov

sethkfman committed:
added exclusions for mocks, util/test, util/testUtils

Gudahtt reviewed (approved):
LGTM!
